Network/src/Server/server_loop.c

```diff
@@ -66,6 +70,49 @@ static void push_command(client_t *client, char *buffer)
     my_free_array(commands_arr);
 }
 
+static void exec_one_gui_command(client_t *client, server_t *server,
+    char *command)
+{
+    char **command_args = my_str_to_word_array(command, " ");
+    int len = my_array_len(command_args);
+
+    for (int j = 0; commands_gui[j].name != NULL; j++) {
+        if (strcmp(command, commands_gui[j].name) == 0 &&
+        len == 1 + commands_gui[j].nb_args) {
+            commands_gui[j].function(client, server, command_args + 1);
+            my_free_array(command_args);
+            break;
+        }
+    }
+    send(client->socket, "ko\n", 3, 0);
+    my_free_array(command_args);
+}
+
+static void exec_gui_commands(client_t *client, server_t *server, char *buffer)
+{
+    char **commands_arr = my_str_to_word_array(buffer, "\n");
+
+    for (int i = 0; commands_arr[i] != NULL; i++)
+        exec_one_gui_command(client, server, commands_arr[i]);
+    my_free_array(commands_arr);
+}
+
+static void client_state_switch(client_t *client, server_t *server,
+    char *buffer)
+{
+    switch (client->state) {
+        case WAITING:
+            start_communication_with_client(client, server, buffer);
+            break;
+        case PLAYING:
+            push_command(client, buffer);
+            break;
+        case GRAPHIC:
+            exec_gui_commands(client, server, buffer);
+            break;
```

Comment:
add a default case to handle potential bug

```suggestion
        case WAITING:
            start_communication_with_client(client, server, buffer);
            break;
        case PLAYING:
            push_command(client, buffer);
            break;
        case GRAPHIC:
            exec_gui_commands(client, server, buffer);
            break;
        default:
            printf("error unknown state");
            break;
```

Reply:
mb, ill take your suggestion